frontend/src/pages/AddFlightPage/AddFlightPage.tsx

```diff
@@ -0,0 +1,60 @@
+import './AddFlightPage.css';
+import FlightForm from "./components/FlightForm.tsx";
+import {NewFlight} from "../../types/model/dataType.ts";
+import {FormEvent, useState} from "react";
+import axios from "axios";
+import {NavigateFunction, useNavigate} from "react-router-dom";
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import {Box} from "@mui/material";
+
+type AddFlightPageProps = {
+    fetchAllFlights: () => void
+}
+
+export default function AddFlightPage({ fetchAllFlights }: Readonly<AddFlightPageProps>) {
+    const [newFlight, setNewFlight] = useState<NewFlight>({
+        flightCode: "",
+        airline: "",
+        origin: "",
+        destination: "",
+        departureTime: "",
+        arrivalTime: "",
+        aircraftType: "",
+        flightStatus: "UNKNOWN"
+    });
+
+    const navigate: NavigateFunction = useNavigate();
+
+    const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        console.log("Data submitted!");
+        axios.post("/api/flight", newFlight)
+            .then(response => console.log(response))
+            .then(() => fetchAllFlights())
+            .then(error => console.log(error));
+
+        navigate("/flight")
+    }
```

Comment:
I recommend to put this into the form component. If u do that, u wont need to pass props and u wont need another useState... u can just take the data from the inputs and send the new flight to the backend